### PR TITLE
Propogate changes to nginx-sm

### PR DESCRIPTION
**Pull request by Rohith-Raju**
**Description**

This PR fixes #
Propagation changes made from [this](https://github.com/meshery/meshery-traefik-mesh/pull/219/files)

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: nginx/nginx.go [hunk 1]
```diff
@@ -187,11 +188,11 @@ func (nginx *Nginx) CreateKubeconfigs(kubeconfigs []string) error {
...
+func (nginx *Nginx) streamErr(summary string, e *meshes.EventsResponse, err error) {
 	e.Summary = summary
 	e.Details = err.Error()
 	e.ErrorCode = errors.GetCode(err)
 	e.ProbableCause = errors.GetCause(err)
 	e.SuggestedRemediation = errors.GetRemedy(err)
 	nginx.StreamErr(e, err)
-}
\ No newline at end of file
+}p
```
Comment (MUzairS15) [inline nginx/nginx.go:198]: Remove this

File: nginx/oam.go [hunk 2]
```diff
@@ -73,29 +76,48 @@ type CompHandler func(*Nginx, v1alpha1.Component, bool, []string) (string, error
...
+			OperationId:   uuid.New().String(),
+			Component:     config.ServerConfig["type"],
+			ComponentName: config.ServerConfig["name"],
+		}
 		fnc, ok := compFuncMap[comp.Spec.Type]
 		if !ok {
 			msg, err := handleNginxCoreComponents(nginx, comp, isDel, "", "", kubeconfigs)
 			if err != nil {
+				ee.Summary = fmt.Sprint("Error while %s %s", stat1, comp.Spec.Type)
+				mesh.streamErr(ee.Summary, ee, err)
```
Comment (MUzairS15) [inline nginx/oam.go:100]: mesh?? mesh --> nginx

File: nginx/oam.go [hunk 3]
```diff
@@ -73,29 +76,48 @@ type CompHandler func(*Nginx, v1alpha1.Component, bool, []string) (string, error
...
+			ee.Details = fmt.Sprintf("The %s is now %s.", comp.Spec.Type, stat2)
+			mesh.StreamInfo(ee)
 			msgs = append(msgs, msg)
 			continue
 		}
 
 		msg, err := fnc(nginx, comp, isDel, kubeconfigs)
 		if err != nil {
+			ee.Summary = fmt.Sprintf("Error while %s %s", stat1, comp.Spec.Type)
+			mesh.streamErr(ee.Summary, ee, err)
```
Comment (MUzairS15) [inline nginx/oam.go:114]: here also same change, mesh --> nginx

**Review by MUzairS15 — CHANGES_REQUESTED** [review]
@Rohith-Raju Try building locally first before committing, you will get all the error and you can make appropriate changes.

**Comment by leecalcote** [comment]
@Rohith-Raju If you haven’t built antacid these changes before submitting for others to spend their time reviewing… It would seem that you have, yet again, not done your homework but are asking others to do it for you.

Perhaps I’m misunderstanding the situation, but given the number of times that we have run into the situation I don’t believe that the benefit of the doubt is appropriate. Just like it’s not appropriate to knowingly turn in an incomplete assignment with the expectation that others carry it to completion.

**Comment by Rohith-Raju** [comment]
@leecalcote truly sorry, won't happen again. Thought this was a plug-and-play kinda thing, but it is not. I should've known better.I'l l compile and make changes accordingly.

**Comment by leecalcote** [comment]
I hear that and if there is any type of issue to learn on, this is a great type to do so with. My ask is that you be upfront with me maintainers, set appropriate expectations and both do your homework and document your studies as proof of diligence.

**Comment by Rohith-Raju** [comment]
@leecalcote, Yes, I'll redo the entire changes

**Comment by gr455** [comment]
@Rohith-Raju there are only very small changes required after this PR, I can make them and keep your commits so you get the credit for your work. On to the next one for you :smile:

**Comment by Rohith-Raju** [comment]
@gr455 sounds good, I'm more than happy to make changes if you point them out :smile:

**Comment by gr455** [comment]
> @gr455 sounds good, I'm more than happy to make changes if you point them out so that I'd know where I went wrong. smile

Oh, @MUzairS15 has pointed out the problems. Very small changes though, I did them. No problem. Thanks for doing the major part :smile:

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**